testing/tests/DevExpress.ui.widgets.dataGrid/editing.tests.js

```diff
@@ -3200,6 +3200,26 @@ QUnit.test('Remove the inserted row with edit mode batch and hidden column', fun
     assert.ok(!testElement.find('tbody > tr').first().hasClass('dx-row-inserted'), 'not has row inserted');
 });
 
+QUnit.test('AddRow method should return Deferred', function(assert) {
+    // arrange
+    this.options.editing = {
+        mode: 'batch',
+        allowAdding: true
+    };
+
+    this.rowsView.render($('#container'));
+
+    // assert
+    assert.equal(this.getVisibleRows().length, 7, '7 visible rows');
+
+    // act
+    this.addRow().done(() => {
+        // assert
+        assert.ok(true, 'addRow returns Deferred');
```

Comment:
The test will be passed if done is not called

Reply:
Done